src/components/common/store/searchReducer.tsx

```diff
@@ -275,6 +284,55 @@ const processDatasetDownload = createAsyncThunk<
   }
 );
 
+const processWFSDownload = createAsyncThunk<
+  ReadableStream,
+  WFSDownloadRequest,
+  { rejectValue: ErrorResponse }
+>(
+  "download/downloadWFS",
+  async (request: WFSDownloadRequest, thunkAPI: any) => {
+    try {
+      // Extract download conditions
+      const dateRange = getDateConditionFrom(request.downloadConditions);
+      const multiPolygon = getMultiPolygonFrom(request.downloadConditions);
+      const format = getFormatFrom(request.downloadConditions); // Currently unused - CSV only for WFS
+
+      const requestBody = {
+        inputs: {
+          uuid: request.uuid,
+          start_date: dateRange.start,
+          end_date: dateRange.end,
+          multi_polygon: multiPolygon,
+          layer_name: request.layerName,
+        },
+        outputs: {},
+        subscriber: {
+          successUri: "",
+          inProgressUri: "",
+          failedUri: "",
+        },
+      };
+
+      return axios
+        .post("/api/v1/ogc/processes/downloadWfs/execution", requestBody, {
+          adapter: "fetch", // Use fetch adapter for streaming
+          responseType: "stream",
+          headers: {
+            "Content-Type": "application/json",
+            Accept: "text/event-stream",
+            "Cache-Control": "no-cache",
+          },
+          timeout: WFS_DOWNLOAD_TIMEOUT, // 20 minutes for WFS downloads
+          signal: thunkAPI.signal,
+        })
+        .then((response) => response.data)
```

Comment:
I think you should return promise directly

      return axios
        .post("/api/v1/ogc/processes/downloadWfs/execution", requestBody, {
          adapter: "fetch", // Use fetch adapter for streaming
          responseType: "stream",
          headers: {
            "Content-Type": "application/json",
            Accept: "text/event-stream",
            "Cache-Control": "no-cache",
          },
          timeout: WFS_DOWNLOAD_TIMEOUT, // 20 minutes for WFS downloads
          signal: thunkAPI.signal,
        })

the caller will be const v = dispatch(processWFSDownload...)


and then you can use v.abort() to cancel the call 

or v.then() to extract the value.